lighthouse-core/audits/seo/crawlable-anchors.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether anchors have hyperlinks which can be crawled by search engines. This descriptive title is shown when all hyperlinks on the page are crawlable. */
+  title: 'Anchors have crawlable hyperlinks',
+  /** Descriptive title of a Lighthouse audit that provides detail on whether anchors have hyperlinks which can be crawled by search engines. This descriptive title is shown when there are hyperlinks which are not crawlable by search engines. */
+  failureTitle: 'Anchors do not have crawlable hyperlinks',
+  /** Description of a Lighthouse audit that tells the user why hyperlinks should be crawlable. This is displayed after a user expands the section to see more. 'Learn More' becomes link text to additional documentation. */
+  description: 'Search engines use hyperlinks to crawl websites',
+  /** Label for a column in a data table; entries will be the HTML anchor elements that failed the audit. Anchors are DOM elements that are links. */
+  columnFailingAnchors: 'Failing Anchor Elements',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class CrawlableAnchors extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'crawlable-anchors',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['AnchorElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit({AnchorElements: anchorElements}) {
+    const failingAnchors = anchorElements.filter(({rawHref, listeners = [], name = ''}) => {
+      if (rawHref.startsWith('file:')) {
+        return true;
+      }
+
+      const hasClickHandler = listeners.some(({type}) => type === 'click');
+
+      if (hasClickHandler || name.trim().length > 0) return;
+
+      if (rawHref === '') {
+        return true;
+      }
+
+      if (rawHref.startsWith('javascript:void(0)')) {
```

Comment:
feels like we should be flexible here w.r.t whitespace, optional parens around `0`, etc

maybe something like? (untested, should probably extract it to a var and all that jazz 😃)

```suggestion
      if (/^javascript:\s*void(\(|\s+)0(\)|\s+)/.test(rawHref)) {
```

Reply:
You're throwing me a lifeline here! I'm no regex wizard so this is helpful. Regarding the regex's, I'm going to comment on my own latest set of changes to highlight a potential issue, so when you see it, have a think please, and thanks again for this one